config/src/shared/connection.rs

```diff
@@ -6,6 +6,112 @@ use tokio_postgres::{Config as TokioPgConnectOptions, config::SslMode as TokioPg
+/// session-specific settings that affect how the server processes queries and data.
+#[derive(Debug, Clone)]
+pub struct PgConnectionOptions {
+    /// Sets the display format for date values.
+    pub datestyle: String,
+    /// Sets the display format for interval values.
+    pub intervalstyle: String,
+    /// Controls the number of digits displayed for floating-point values.
+    pub extra_float_digits: i32,
+    /// Sets the client-side character set encoding.
+    pub client_encoding: String,
+    /// Sets the time zone for displaying and interpreting time stamps.
+    pub timezone: String,
+    /// Aborts any statement that takes more than the specified number of milliseconds.
+    pub statement_timeout: u32,
+    /// Aborts any statement that waits longer than the specified milliseconds to acquire a lock.
+    pub lock_timeout: u32,
+    /// Terminates any session that has been idle within a transaction for longer than the specified milliseconds.
+    pub idle_in_transaction_session_timeout: u32,
+    /// Sets the application name to be reported in statistics views and logs for connection identification.
+    pub application_name: String,
+}
+
+impl Default for PgConnectionOptions {
+    /// Returns default configuration values optimized for ETL/replication workloads.
+    ///
+    /// These defaults ensure consistent behavior across different PostgreSQL installations
+    /// and are specifically tuned for ETL systems that perform logical replication and
+    /// large data operations:
+    ///
+    /// - `datestyle = "ISO"`: Provides consistent date formatting for reliable parsing
+    /// - `intervalstyle = "postgres"`: Uses standard PostgreSQL interval format
+    /// - `extra_float_digits = 3`: Ensures sufficient precision for numeric replication
+    /// - `client_encoding = "UTF8"`: Supports international character sets
+    /// - `timezone = "UTC"`: Eliminates timezone ambiguity in distributed ETL systems
+    /// - `statement_timeout = 7200000` (2 hours): Allows large COPY operations while preventing runaway queries
+    /// - `lock_timeout = 30000` (30 seconds): Prevents indefinite blocking on table locks during replication
+    /// - `idle_in_transaction_session_timeout = 300000` (5 minutes): Cleans up abandoned transactions that could block VACUUM
+    /// - `application_name = "etl"`: Enables easy identification in monitoring and pg_stat_activity
+    fn default() -> Self {
+        Self {
+            datestyle: "ISO".to_string(),
+            intervalstyle: "postgres".to_string(),
+            extra_float_digits: 3,
+            client_encoding: "UTF8".to_string(),
+            timezone: "UTC".to_string(),
+            statement_timeout: 7_200_000, // 2 hours in milliseconds
+            lock_timeout: 30_000,         // 30 seconds in milliseconds
+            idle_in_transaction_session_timeout: 300_000, // 5 minutes in milliseconds
+            application_name: "etl".to_string(),
```

Comment:
What does Postgres to Postgres replication use for the timeout settings? Ok to use `etl` as application name for now, but we should later on use separate names for table sync and apply workers, probably same or similar as slot names.

Reply:
Maybe we should enable users to configure it in the future and keep it disabled for now, this way we can avoid possible problems related to timeouts.

Reply:
Let's disable `statement_timeout` and can keep the other timeouts at their current values in the PR.

Reply:
By disable I meant that `statement_timeout` is set to 0. If we omit it it will take the default value set for the role we use to connect which may or may not be 0.